tests/BuildSystem/Build/directory-tree-structure-signatures.llbuild

```diff
@@ -0,0 +1,100 @@
+# Check the handling of directory tree structure signatures.
+#
+# We check the node used against three kinds of directories: missing, a file, an
+# actual directory.
+#
+# RUN: rm -rf %t.build
+# RUN: mkdir -p %t.build
+# RUN: cp %s %t.build/build.llbuild
+# RUN: echo "file" > %t.build/file
+# RUN: %{llbuild} buildsystem build --serial --chdir %t.build > %t.initial.out
+# RUN: %{FileCheck} --check-prefix=CHECK-INITIAL --input-file=%t.initial.out %s
+#
+# CHECK-INITIAL: DTS-MISSING-CHANGED
+# CHECK-INITIAL: DTS-FILE-CHANGED
+# CHECK-INITIAL: DIR-CREATOR
+# CHECK-INITIAL: DTS-DIR-CHANGED
+
+
+# Check that a null build does nothing.
+#
+# RUN: echo "START." > %t.rebuild.out
+# RUN: %{llbuild} buildsystem build --serial --chdir %t.build >> %t.rebuild.out
+# RUN: echo "EOF" >> %t.rebuild.out
+# RUN: %{FileCheck} --check-prefix=CHECK-REBUILD --input-file=%t.rebuild.out %s
+#
+# CHECK-REBUILD: START
+# CHECK-REBUILD-NOT: DTS-
+# CHECK-REBUILD-NEXT: EOF
+
+
+# Check that modifications are detected.
+#
+# RUN: mkdir -p %t.build/missing-dir
+# RUN: mkdir -p %t.build/dir/subdir
+# RUN: echo "created" > %t.build/dir/subdir/file
+# RUN: echo "modified" > %t.build/file
+# RUN: %{llbuild} buildsystem build --serial --chdir %t.build > %t.modified.out
+# RUN: %{FileCheck} --check-prefix=CHECK-MODIFIED --input-file=%t.modified.out %s
+#
+# CHECK-MODIFIED: DTS-MISSING-CHANGED
+# CHECK-MODIFIED: DTS-FILE-CHANGED
+# CHECK-MODIFIED: DTS-DIR-CHANGED
+
+
+# Check that a mutation of a file is ignored.
+#
+# RUN: echo "mutated" >> %t.build/dir/subdir/file
+# RUN: %{llbuild} buildsystem build --serial --chdir %t.build > %t.mutated.out
+# RUN: echo EOF. >> %t.mutated.out
+# RUN: %{FileCheck} --check-prefix=CHECK-MUTATED --input-file=%t.mutated.out %s
+#
+# CHECK-MUTATED-NOT: DTS-MISSING-CHANGED
+# CHECK-MUTATED-NOT: DTS-FILE-CHANGED
+# CHECK-MUTATED-NOT: DTS-DIR-CHANGED
+# CHECK-MUTATED: EOF.
+
+client:
+  name: basic
+
+targets:
+  "": ["<all>"]
+
+nodes:
+  "missing-dir/":
+    is-directory-structure: true
+  "file/":
+    is-directory-structure: true
+  "dir/":
+    is-directory-structure: true
```

Comment:
Random question: What if I also need `dir/` as a node with is-directory-structure = false?

Reply:
That may not work out currently... we don't yet have a way to override the target path independent of the name, so unfortunately it has to be mangled this way. I could probably make it work to use "dir//" as a workaround.